tests/unit/models/test_gpt2.py

```diff
@@ -106,3 +106,11 @@ def should_work_on_multiple_sentences():
                 assert math.isclose(score, exp_score, rel_tol=eps), {"index": (i, j)}
             assert ids == exp_ids
             assert tokens == exp_tokens
+
+    @pytest.mark.xfail
+    def should_work_with_a_sentence_longer_than_the_model_max_size():
+        max_input_size = scorer.tokenizer.max_len
+        long_sentence = ["The " * (max_input_size + 1)]
+        warning = "One of the sentences is longer than the model max size."
+        with pytest.warns(UserWarning, warning):
```

Comment:
Does the transformers library rise this warning?

Reply:
No, as I said, transformers raised an RunTimeError. I am not sure to understand what you want to achieve with this xfail.test.

Reply:
Just a test like this one:
https://github.com/simonepri/lm-scorer/blob/master/tests/unit/models/test_gpt2.py#L41-L66
Where we assert that both the ids and the tokens returned by the function are the one expected (we can skip to check wether the scores are right because we don't know them).

The test basically describes what should be the correct behavior of the function. Our implementation currently does not behave like this and will make the test fail. The xfail.test tells pytest that it is okay that the test fails.

Reply:
```
    def should_work_with_a_sentence_longer_than_the_model_max_size():
        max_input_size = scorer.tokenizer.max_len
        long_sentence = "Very" + " long" * max_input_size
        eps = 1e-4
        exp_ids = [16371] + [809] * max_input_size + [50256]
        exp_tokens = ["Very"] + ["Ġlong"] * max_input_size + ["<|endoftext|>"]

        _, ids, tokens = scorer._tokens_log_prob_for_batch([long_sentence])[0]
        ids = ids.tolist()

        assert ids == exp_ids
        assert tokens == exp_tokens
```
Ok like that ?

Reply:
Yes perfect, just remove `eps`.